unittests/test_cli.py

```diff
@@ -627,6 +627,23 @@ def test_use_module_path(run_reframe, user_exec_ctx):
     assert returncode == 0
 
 
+def test_overwrite_module_path(run_reframe, user_exec_ctx):
+    ms = rt.runtime().modules_system
+    if ms.name == 'nomod':
+        pytest.skip('no modules system found')
+
+    module_path = 'unittests/modules'
+    returncode, stdout, stderr = run_reframe(
+        more_options=[f'--module-path={module_path}', '-m testmod_foo'],
+        config_file=fixtures.USER_CONFIG_FILE, action='run',
+        system=rt.runtime().system.name
+    )
+
+    assert 'Traceback' not in stdout
+    assert 'Traceback' not in stderr
```

Comment:
I think you should also add an assertion here that you didn't get the warning that `testmod_foo` could not be loaded. See here:

https://github.com/eth-cscs/reframe/blob/4aa8ac9029e9d55bc4e2283c6c16fcb4ba6c54dd/reframe/frontend/cli.py#L741-L744